enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx

```diff
@@ -0,0 +1,86 @@
+import { useMemo } from "react";
+import { P, match } from "ts-pattern";
+
+import {
+  InteractiveDashboard,
+  InteractiveQuestion,
+  defineMetabaseAuthConfig,
+} from "embedding-sdk";
+import { MetabaseProvider } from "embedding-sdk/components/public/MetabaseProvider";
+import { Box } from "metabase/ui";
+import { hasPremiumFeature } from "metabase-enterprise/settings";
+
+import { useSdkIframeEmbedEventBus } from "../hooks/use-sdk-iframe-embed-event-bus";
+import type { SdkIframeEmbedSettings } from "../types/embed";
+
+import { SdkIframeInvalidLicenseError } from "./SdkIframeError";
+
+export const SdkIframeEmbedRoute = () => {
+  const { embedSettings } = useSdkIframeEmbedEventBus();
+
+  const authConfig = useMemo(() => {
+    if (!embedSettings) {
+      return;
+    }
+
+    // TODO: add support for SSO auth once the new SSO implementation on the SDK is ready
+    if (!embedSettings.instanceUrl || !embedSettings.apiKey) {
+      return;
+    }
+
+    return defineMetabaseAuthConfig({
+      metabaseInstanceUrl: embedSettings.instanceUrl,
+      apiKey: embedSettings.apiKey,
+    });
+  }, [embedSettings]);
+
+  // The SDK will show its own loading indicator, so we don't need to show it twice.
+  if (embedSettings === null || !authConfig) {
```

Comment:
it feels like we're mixing validation and the definition of `authConfig`. What about we check everything upfront, `embeddingSettings` couldn't be `null` or you could use `!embedSettings` like line 22. And check `!embedSettings.instanceUrl || !embedSettings.apiKey)` here too.

That way, we have a single location that returns `null`, well... you also have a single null return location, but `authConfig` also partially do validation, and I'd rather we have everything in one place, so I don't have to jump around when trying to understand this component.

Reply:
That works! Let me move it up.

Reply:
Thanks! Fixed in https://github.com/metabase/metabase/pull/56915/commits/0d6440e6b51125b2756143710fce7e6ddca10490. The reason that condition was needed was because of `authConfig` being `useMemo`'d, but I think we actually might _not_ need to memoize it from what I can see in `use-init-data.ts`.

Reply:
this is much cleaner!